packages/terra-embedded-content-consumer/src/EmbeddedContentConsumer.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { Consumer } from 'xfc';
+import styles from './EmbeddedContentConsumer.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The source URL of the content to load.
+   */
+  src: PropTypes.string.isRequired,
+  /**
+   * Notifies the component that the container has been launched.
+   */
+  onLaunch: PropTypes.func,
+  /**
+   * Notifies the component that the container has been authorized.
+   */
+  onAuthorize: PropTypes.func,
+  /**
+   * The component can be configured with an authorization secret.
+   * secret - The authorization secret to be used if the embedded application does not know which domain to trust.
+   */
+  options: PropTypes.shape({
+    secret: PropTypes.string,
+  }),
+  /**
+  * A set of event handlers keyed by the event name.
+  */
+  eventHandlers: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string,
+    handler: PropTypes.func,
+  })),
+  /**
+   * Whether or not the container should be expanded to fill its parent.
+   */
+  fill: PropTypes.bool,
+};
+
+class EmbeddedContentConsumer extends React.Component {
+  componentDidMount() {
+    // Mount the provided source as the application into the content wrapper.
+    this.xfcFrame = Consumer.mount(this.embeddedContentWrapper, this.props.src, this.props.options);
```

Comment:
This will sound more like a comment for xfc. I noticed that here we mount frame first and then add listeners to the frame. It should be fine in most cases since mounting will take some time to complete, however, events can potentially be triggered before event listeners get added. I wish we can pass event handlers as part of options to Consumer's `mount` method. Thoughts? @mhemesath @bjankord

Reply:
What is the intended workflow to initialize an xfc frame if not for mounting it? Initializing the consumer just defaults the globalHandlers and mount returns the frame after emitting the xfc.mounted event. I agree that it would have been helpful to mount a frame with event handlers.

Reply:
I agree... event handlers should be able to be passed ahead of mount